subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftCachingIntegrationTest.groovy

```diff
@@ -0,0 +1,162 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibraries
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Unroll
+
+@Requires(TestPrecondition.SWIFT_SUPPORT)
+class SwiftCachingIntegrationTest extends AbstractInstalledToolChainIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    def app = new SwiftAppWithLibraries()
+
+    def setupProject(TestFile project = temporaryFolder.testDirectory) {
+        project.file('settings.gradle') << '''
+            include 'hello', 'log'
+            rootProject.name = 'app'
+        '''
+        project.file('settings.gradle') << localCacheConfiguration()
+        project.file('build.gradle').text = '''
+            apply plugin: 'swift-executable'
+            dependencies {
+                implementation project(':hello')
+            }
+
+            project(':hello') {
+                apply plugin: 'swift-library'
+                dependencies {
+                    api project(':log')
+                }
+            }
+            project(':log') {
+                apply plugin: 'swift-library'
+            }
+        '''
+        app.logLibrary.writeToProject(project.file('log'))
+        app.library.writeToProject(project.file('hello'))
+        app.executable.writeToProject(project)
+    }
+
+    def 'compilation can be cached'() {
+        setupProject()
+        def allCompileTasks = ['', ':hello', ':log'].collect { compileTask(it, buildType) } as String[]
+
+        when:
+        withBuildCache().run compileTask(buildType)
+
+        then:
+        executedAndNotSkipped allCompileTasks
+
+        when:
+        withBuildCache().run 'clean', installTask(buildType)
+
+        then:
+        skipped allCompileTasks
+        installation(installDir(buildType)).exec().out == app.expectedOutput
+
+        where:
+        buildType << ['Debug', 'Release']
+    }
+
+    @Unroll
+    def "compilation task is relocatable (#buildType)"() {
+        def originalLocation = file('original-location')
+        def newLocation = file('new-location')
+        def allCompileTasks = ['', ':hello', ':log'].collect { compileTask(it, buildType) } as String[]
+        setupProject(originalLocation)
+        setupProject(newLocation)
+
+        when:
+        inDirectory(originalLocation)
+        withBuildCache().run compileTask(buildType)
```

Comment:
I still think we should be explicit about `succeeds`. This would work even if the build failed in some weird way.

Reply:
`run` also asserts success: https://github.com/gradle/gradle/blob/8c455eb91a1870fd2e21caa49a8cf696d40aedc5/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy#L159-L161

Reply:
Ha! I didn't know that. Thanks :)